src/interchange/src/avro/encode.rs

```diff
@@ -213,6 +221,16 @@ impl Encoder {
     }
 }
 
+impl Encode for AvroEncoder {
+    fn encode_key_unchecked(&self, row: Row) -> Vec<u8> {
+        self.encode_key_unchecked(self.key_schema_id.unwrap(), row)
```

Comment:
I don't like the `unwraps`, but getting rid of them seems even more complicated.

Reply:
Agreed! I'm going to play with making a second Encoder, which might help.